imap_processing/hit/l1b/hit_l1b.py

```diff
@@ -0,0 +1,165 @@
+"""IMAP-HIT L1B data processing."""
+
+import logging
+from dataclasses import fields
+
+import numpy as np
+import xarray as xr
+
+from imap_processing import utils
+from imap_processing.cdf.global_attrs import ConstantCoordinates
+from imap_processing.cdf.utils import write_cdf
+from imap_processing.hit import hit_cdf_attrs
+from imap_processing.hit.l0.data_classes.housekeeping import Housekeeping
+
+logger = logging.getLogger(__name__)
+
+# TODO review logging levels to use (debug vs. info)
+
+
+def hit_l1b(l1a_dataset: xr.Dataset):
+    """Process HIT data to L1B.
+
+    Parameters
+    ----------
+    l1a_dataset : xarray.Dataset
+        l1A data
+
+    Returns
+    -------
+    xarray.Dataset
+        L1B processed data
+    """
+    # TODO: Check for type of L1A dataset and determine what L1B products to make
+    #   Need more info from instrument teams. Work with housekeeping data for now
+    logical_source = "imap_hit_l1b_hk"
+
+    # Create datasets
+    datasets = []
+    if "_hk" in logical_source:
+        dataset = create_hk_dataset()
+        datasets.append(dataset)
+
+    # Create CDF files
+    logger.info("Creating CDF files for HIT L1B data")
+    cdf_filepaths = []
+    for dataset in datasets:
+        cdf_file = write_cdf(dataset)
+        cdf_filepaths.append(cdf_file)
+    logger.info(f"L1B CDF files created: {cdf_filepaths}")
+    return cdf_filepaths
+
+
+# TODO: This is going to work differently when we have sample data
+def create_hk_dataset():
+    """Create a housekeeping dataset.
+
+    Returns
+    -------
+    xr.dataset
+        Dataset with all data product fields in xr.DataArray
+    """
+    logger.info("Creating datasets for HIT L1B data")
+
+    # TODO: TEMPORARY. Need to update to use the L1B data class once that exists.
+    #  Using l1a housekeeping data class for now since l1b housekeeping has the
+    #  same data fields
+    data_fields = fields(Housekeeping)
+
+    # TODO define keys to skip. This will change later.
+    skip_keys = [
+        "shcoarse",
+        "ground_sw_version",
+        "packet_file_name",
+        "ccsds_header",
+        "leak_i_raw",
+    ]
+
+    # Create fake data for now
+
+    # Convert integers into datetime64[s]
+    epoch_converted_time = [utils.calc_start_time(time) for time in [0, 1, 2]]
+
+    # Create xarray data arrays for dependencies
+    epoch_time = xr.DataArray(
+        data=epoch_converted_time,
+        name="epoch",
+        dims=["epoch"],
+        attrs=ConstantCoordinates.EPOCH,
+    )
+
+    adc_channels = xr.DataArray(
```

Comment:
Should change this to 64 instead of 3

Reply:
Maybe make it a variable to use in your shape.size creation below as well?
`nchannels = 64` ... `np.ones((nepochs, nchannels))`